src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -258,28 +259,88 @@ def __len__(self) -> int:
         """
         return len(self._buffer)
 
-    @overload
-    def __getitem__(self, key: SupportsIndex) -> float:
-        """See the main __getitem__ method.
+    def window(
+        self,
+        start: int | datetime,
+        end: int | datetime,
+        fill_gaps: Optional[float] = None,
```

Comment:
Are there really gaps or only the single, initially uninitialized part of the memory?